osu.Game/Rulesets/Mods/ModHealthChallenge.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Scoring;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public class ModHealthChallenge : ModChallenge, IApplicableToHealthProcessor
+    {
+        public override string Name => "Health Challenge";
+        public override string Acronym => "HC";
+        public override string Description => "Hold your health or you fail";
+
+        [SettingSource("Health Threshold", "Set your own health threshold")]
+        public Bindable<double> HealthThreshold { get; } = new BindableDouble
+        {
+            Precision = 0.01,
+            MinValue = 0.50,
+            MaxValue = 1.00,
+            Default = 0.96,
+            Value = 0.96,
+        };
+
+        public override string SettingDescription => $"{HealthThreshold.Value:0%}";
+
+        public void ApplyToHealthProcessor(HealthProcessor healthProcessor)
+        {
+            healthProcessor.FailConditions += checkThresholdCondition;
+        }
+
+        private bool checkThresholdCondition(HealthProcessor healthProcessor, JudgementResult result)
+        {
+            // Need to manually checked here since HasCompleted is updated after ApplyResultInternal
+            bool allJudged = healthProcessor.JudgedHits == healthProcessor.MaxHits;
+
+            if (!allJudged) return false;
+
+            return result.HealthAtJudgement < HealthThreshold.Value;
```

Comment:
Isn't this going to use the previous judgement? The xmldoc for `HealthAtJudgement` states "health prior to this <see cref="JudgementResult"/> occurring.". Maybe you should use `Health` instead?

Reply:
Ah right! Will change